libtransmission/crypto-utils.h

```diff
@@ -53,10 +48,19 @@ bool tr_sha1_update(tr_sha1_ctx_t handle, void const* data, size_t data_length);
 /**
  * @brief Finalize and export SHA1 hash, free hasher context.
  */
-bool tr_sha1_final(tr_sha1_ctx_t handle, uint8_t* setme);
-
 std::optional<tr_sha1_digest_t> tr_sha1_final(tr_sha1_ctx_t handle);
 
+/**
+ * @brief Generate a SHA1 hash from one or more chunks of memory.
+ */
+template<typename... T>
+std::optional<tr_sha1_digest_t> tr_sha1(T... args)
+{
+    auto ctx = tr_sha1_init();
```

Comment:
`tr_sha1_init()` may return `nullptr`.

Reply:
fixed in 5ae17bbe1031636fd89d23423f0891381df44006